src/python/pants/backend/go/util_rules/pkg_pattern.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
This file is adapted from `go` toolchain source. It is broken out as a separate commit in this PR.